src/promptflow/promptflow/azure/operations/_flow_operations.py

```diff
@@ -54,6 +71,182 @@ def __init__(
         self._service_caller = service_caller
         self._credential = credential
 
+    @cached_property
+    def _common_azure_url_pattern(self):
+        operation_scope = self._operation_scope
+        url = (
+            f"/subscriptions/{operation_scope.subscription_id}"
+            f"/resourceGroups/{operation_scope.resource_group_name}"
+            f"/providers/Microsoft.MachineLearningServices"
+            f"/workspaces/{operation_scope.workspace_name}"
+        )
+        return url
+
+    def _get_flow_portal_url(self, flow_resource_id: str):
+        """Get the portal url for the run."""
+        match = self._FLOW_RESOURCE_PATTERN.match(flow_resource_id)
+        if not match or len(match.groups()) != 2:
+            logger.warning("Failed to parse flow resource id '%s'", flow_resource_id)
+            return None
+        datastore, flow_id = match.groups()
+        url = f"https://ml.azure.com/prompts/flow/{datastore}/{flow_id}/details?wsid={self._common_azure_url_pattern}"
+        return url
+
+    @monitor_operation(activity_name="pfazure.flows.create_or_update", activity_type=ActivityType.PUBLICAPI)
+    def create_or_update(self, flow: Union[str, Path], name=None, type=None, **kwargs) -> Flow:
+        """Create a flow to remote from local source.
+
+        :param flow: The source of the flow to create.
+        :type flow: Union[str, Path]
+        :param name: The name of the flow to create. Default to be flow folder name + timestamp if not specified.
+            e.g. "web-classification-10-27-2023-14-19-10"
+        :type name: str
+        :param type: The type of the flow to create. One of ["standard", evaluation", "chat"].
+            Default to be "standard" if not specified.
+        :type type: str
+        :param description: The description of the flow to create. Default to be the description in flow yaml file.
+        :type description: str
+        :param tags: The tags of the flow to create. Default to be the tags in flow yaml file.
+        :type tags: Dict[str, str]
+        """
+        # validate the parameters
+        azure_flow, flow_name, flow_type, kwargs = self._validate_flow_creation_parameters(flow, name, type, **kwargs)
+        # upload to file share
+        file_share_flow_path = self._resolve_flow_code_and_upload_to_file_share(flow=azure_flow, flow_name=flow_name)
+        if not file_share_flow_path:
+            raise FlowOperationError(f"File share path should not be empty, got {file_share_flow_path!r}.")
```

Comment:
Seems this should be child of UserErrorException?

Reply:
This is used for general flow operation error now, seems most of our cli/sdk errors are not having a clear user/system error definition now. May need to improve in a separate item.